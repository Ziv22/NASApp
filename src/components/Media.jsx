import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';

export default function Media(props) {
    const [videoSize , setVideoSize] = useState(0)

    useEffect(()=>{
        (   async ()=>{

            }
        )()   
    },[])

    if(props.location === "/Favourites"){
        return (
        <Link to={`/Favourites/${props.id}`}>
            <img className="card-image" src={props.imgUrl} alt=""/>
        </Link>
        )
    } else if(props.location === "/Home"){
        return (
        <iframe className="video-image" src={props.imgUrl}></iframe>
        )
    }else{
        return <img className="card-image" src={props.imgUrl} alt=""/>
    }
}
