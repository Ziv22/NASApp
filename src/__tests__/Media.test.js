import React from "react";
import Media from "../components/Media";
import renderer from "react-test-renderer";
import { cleanup } from "@testing-library/react";

afterEach(cleanup);
const id = 45;

test("Media with location props equals to /Home and has a link of an image", () => {
  const imgUrl =
    "https://www.sciencenewsforstudents.org/wp-content/uploads/2020/10/102320_MT_moon-brief_feat-1028x579.jpg";
  const location = "/Home";
  const mediaSnapshot = renderer
    .create(<Media id={id} imgUrl={imgUrl} location={location} />)
    .toJSON();
  expect(mediaSnapshot).toMatchSnapshot();
});

test("Media with location props equals to /Home and has a link of a video", () => {
  const imgUrl = "https://www.youtube.com/embed/6AviDjR9mmo";
  const location = "/Home";
  const mediaSnapshot = renderer
    .create(<Media id={id} imgUrl={imgUrl} location={location} />)
    .toJSON();
  expect(mediaSnapshot).toMatchSnapshot();
});

test("Media with no location props", () => {
  const imgUrl =
    "https://www.sciencenewsforstudents.org/wp-content/uploads/2020/10/102320_MT_moon-brief_feat-1028x579.jpg";
  const mediaSnapshot = renderer
    .create(<Media id={id} imgUrl={imgUrl} />)
    .toJSON();
  expect(mediaSnapshot).toMatchSnapshot();
});
