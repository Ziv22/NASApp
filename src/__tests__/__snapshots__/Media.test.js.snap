// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media with location props equals to /Home and has a link of a video 1`] = `
<iframe
  className="video-image"
  src="https://www.youtube.com/embed/6AviDjR9mmo"
/>
`;

exports[`Media with location props equals to /Home and has a link of an image 1`] = `
<img
  alt=""
  className="card-image"
  src="https://www.sciencenewsforstudents.org/wp-content/uploads/2020/10/102320_MT_moon-brief_feat-1028x579.jpg"
/>
`;

exports[`Media with no location props 1`] = `
<img
  alt=""
  className="card-image"
  src="https://www.sciencenewsforstudents.org/wp-content/uploads/2020/10/102320_MT_moon-brief_feat-1028x579.jpg"
/>
`;
